File: api/Models/PreOrderSchedule.cs
using System;

namespace ByzmoApi.Models
{
    public class PreOrderSchedule
    {
        public int Id { get; set; }

        public int OrderId { get; set; }
        public int ProductId { get; set; }
        public string PaymentTerm { get; set; }
        public string PaymentTermDesc { get; set; }
        public decimal Amount { get; set; }
        public bool IsCleared { get; set; }
        public bool IsSendEmail { get; set; }
        public bool IsPrSend { get; set; }
        public PaymentMethod PaymentMethod { get; set; }
        public int PaymongoStatus  { get; set; }
    }
}

File: web/src/app/classes/pre-order-schedule.ts
import { Utils } from '../app.utils';
import { FileViewer } from './file-viewer';
import { PaymentMethod } from './payment-method';

export class PreOrderSchedule {
    id: number;
    orderId: number;
    productId: number;
    paymentTerm: string;
    amount: number;
    isCleared: boolean;
    attachment: FileViewer;
    isSendEmail: boolean;
    isPrSend: boolean;
    paymentMethod: PaymentMethod;
    dpWithRushFee: number;
    paymongoStatus: number;

    constructor() {
        this.attachment = new FileViewer();
    }
    get paymentTermDesc() {
        let desc = '';
        if (!Utils.isNullOrUndefined(this.paymentTerm)) {
            desc = Utils.PREORDER_TERM.filter(x => x.code === this.paymentTerm)[0].description;
        }
        return desc;
    }
    set paymentTermDesc(a) { }
}
